### ISSUE TITLE
Cannot connect to debugger (--inspector flag)

### ISSUE THREAD

**Issue by babakness** [Microsoft/vscode#30439]
- VSCode Version: 1.13.1
- OS Version: 10.12.5
- Node: 8.1.2
- Chrome 59
Steps to Reproduce:

1. `node --inspect=0.0.0.0:9229 index.js`

```
Debugger listening on ws://0.0.0.0:9229/...
For help see https://nodejs.org/en/docs/inspector
```

contents of launch.json
```
{
 
    "version": "0.2.0",
    "configurations": [
      
        {

            "type": "node",
            "protocol": "inspector",
            "request": "attach",
            "name": "Attach to Process",
            "address": "0.0.0.0",
            "port": 9229
        }
    ]
}
```

2. start debugger in VS Code. Stalls for a while then:

`Ensure Node was launched with --inspect. Cannot connect to runtime process, timeout after 10000 ms - (reason: Cannot connect to the target: connect ECONNREFUSED 0.0.0.0:9229).`

NOTE:
Inspector connects when using Chrome, `chrome://inspect` detects the node and stops at break points as expected.

**Comment by roblourens** [comment]
What if you omit the `0.0.0.0`?

**Comment by roblourens** [comment]
This works for me...

Also, can you navigate to `http://0.0.0.0/json` after starting Node?

**Comment by babakness** [comment]
http://0.0.0.0/json (port 80?) returns nothing as expected (since a webserver isn't running at that port)

port 9229, nothing.

Omitting 0.0.0.0 and starting node with just --inspect without specifying IP results in this:

> 
Ensure Node was launched with --inspect. Cannot connect to runtime process, timeout after 10000 ms - (reason: Cannot connect to the target: connect ECONNREFUSED 127.0.0.1:9229).

Note: this is with Node 8 again, Chrome works.

**Comment by roblourens** [comment]
Right I meant :9229, thanks.

Hm... can you try any other combinations, like just `--inspect` and `http://localhost:9229`, or a different port? Or (god forbid) rebooting? I don't know why node would fail to start the webserver. And chrome should be using basically the same method to discover it...

**Comment by babakness** [comment]
@roblourens But Node does start the server and Chrome can connect... VSCode doesn't. I tried all those variations to no avail... VS Code does not work on this version of Node for me... what version of Node are you using? What does your Launch.json look like?

**Comment by roblourens** [comment]
By 'server' I mean node's websocket server to discover debug connections. When you tried `http://0.0.0.0:9229/json` in a browser, does it say it can't find the server, or does it give an empty response or something?

For me this works with any version of node.

**Comment by AndyDBell** [comment]
I got this problem in VSCode Debugger when mistakenly thinking I should run the "Attach" configuration rather than the "Launch" configuration in Launch.json.  (There was no running process to attach to I guess.)   So, after a wait "Ensure Node was launched with --inspect. Cannot connect to runtime process..." message appeared.  When I selected to run the "Launch" configuration everything worked perfectly. Duh!
